internal/worker/clienterrors/errors.go

```diff
@@ -78,6 +82,17 @@ func GetStatusCode(err *Error) StatusCode {
 	}
 }
 
+func (e *Error) HasDependencyError() bool {
+	switch e.ID {
+	case ErrorDepsolveDependency:
+		return true
+	case ErrorManifestDependency:
+		return true
```

Comment:
we'd also need to include the empty manifest here I think.

Maybe it's just safer to always check the status of the dependencies and if the have a JobError keep walking up the graph?